### DISCUSSION TITLE
[Feature] - Helm Chart

### DISCUSSION THREAD

**Discussion by williejay2017** [valkey-io/valkey#338] [General]
Just wondering if there is a helm chart available for this yet? Wanted to do some testing with our api gateway (which currently uses redis) but we ref the upstream redis chart as a dep in _our_ chart repo.

**Comment by hairmare** [comment]
The Argo CD's downstream Helm chart distro could also benefit from this.

Argo CD currently uses Bitnami's Redis chart and Bitnami have [signalized their willingness](https://github.com/valkey-io/valkey/issues/65#issuecomment-2027437711) to provide a ValKey chart. I didn't yet check if anything happened in the meantime tho

**Reply by gtudan** [reply]
Images are finally there! Did somebody here from Bitnami if they plan to add a chart?

**Reply by carrodher** [reply]
Yep, we (Bitnami) are working on the Helm chart right now once the container images were released at DockerHub ([bitnami/valkey](https://hub.docker.com/r/bitnami/valkey) and [bitnami/valkey-sentinel](https://hub.docker.com/r/bitnami/valkey-sentinel)).
Hopefully, it will be released this month.

**Reply by Dmarcotrigiano** [reply]
It was just merged

**Comment by dmitrypol** [comment]
You can also find Valkey images on https://hub.docker.com/r/valkey/valkey/ if you want to use them directly

**Comment by carrodher** [comment]
At https://github.com/bitnami/charts/pull/25643 you can see the PR implementing the Valkey Helm chart

**Comment by carrodher** [comment]
I'm excited to share that we've just released the Bitnami Valkey Helm chart, now available on Docker Hub at https://hub.docker.com/r/bitnamicharts/valkey, simplifying the deployment of Valkey in Kubernetes.

Built upon the same principles as our other Bitnami charts, Valkey follows the latest best practices and security industry standards. Learn more about our commitment to security [here](https://blog.bitnami.com/2024/04/bitnami-helm-charts-are-now-more-secure.html).

We welcome your feedback and contributions on GitHub. Whether it's reporting issues, suggesting improvements, or contributing code, your input is invaluable so we can, together, continue to enhance the product.

**Reply by ikersuen** [reply]
![image](https://github.com/valkey-io/valkey/assets/39717503/71257460-48d8-457a-9686-487bdfdd1bc5)
![image](https://github.com/valkey-io/valkey/assets/39717503/52c14efe-4503-4a9e-8147-80ebd6a56eb6)
Thanks for making valkey helm chart! It would be great if we could enable cluster mode and have "cluster" option added to Valkey architecture! It would be super helpful since some tools only support "cluster" type of redis for HA. Hoping this feature can release soon~ Thanks~~

**Reply by carrodher** [reply]
In the case of Redis, we have two different charts: Redis and Redis Cluster. [Here](https://github.com/bitnami/charts/tree/main/bitnami/redis-cluster#choose-between-redis-helm-chart-and-redis-cluster-helm-chart) you can see the differences. Do you think the same separation could make sense for Valkey?

**Reply by ikersuen** [reply]
I understand that "redis" and "redis-cluster" are two different repositories and approaches. Sorry for the misunderstanding; I was not expecting something like "redis" Helm chart to support architecture similar to "redis-cluster" Helm Chart.

Since Valkey is new, I'm not sure if there will be a new Helm chart called "valkey-cluster" or how it will be managed in the future. From my understanding, Valkey itself supports cluster architecture, but there isn't an existing Helm chart for it yet. So this might be a dumb question, but will there be a "valkey-cluster" Helm chart under development, or is it technically not feasible?

Thanks in advance!

**Reply by jumpyhook** [reply]
Hi @carrodher,

Is there any plans or road map to add clustered support to valkey on bitnami charts?

**Reply by viceice** [reply]
@carrodher Would you accept a PR for a `valkey-cluster` chart? It would mainly a clone of the `redis-cluster` chart.

**Reply by carrodher** [reply]
Hi! Thanks for your suggestions and collaboration offers 😄 We're already working on the `bitnami/valkey-cluster` chart!
We'll notify here once ready so you can give it a try.

**Reply by carrodher** [reply]
[Here](https://github.com/bitnami/charts/pull/27737) you can find the PR where Valkey Cluster is being implemented. Just in case you would like to take a look. Soon it will be ready in DockerHub!

**Reply by carrodher** [reply]
It is live now, see https://hub.docker.com/r/bitnamicharts/valkey-cluster.
You can see the source code of the Helm chart at https://github.com/bitnami/charts/tree/main/bitnami/valkey-cluster.

If you find any issues or suggestions, feel free to create an Issue in the bitnami/charts GH repo.

**Reply by royatanu94** [reply]
Hi,
We used to deploy redis master & sentinel pod with single image but in valkey it seems there are two different image for valkey-master & valkey-sentinel.
I have found valkey docker file : https://github.com/valkey-io/valkey-container
Can you please provide docker file for valkey sentinel
Also please share if you have any doc to migrate valkey-sentinel from redis-sentinel 
TIA

**Comment by jonasled** [comment]
As Bitnami is now moving all production images and charts behind a subscription ( https://github.com/bitnami/charts/issues/35164 ), it would be great, to get a official Helm Chart.

**Reply by majst01** [reply]
Was released under the Apache v2 License, so forking it should be possible

**Reply by muffl0n** [reply]
It would be awesome if this fork could reside under https://github.com/valkey-io

Are there any plans?

**Reply by kimminw00** [reply]
+1

**Reply by pat-s** [reply]
This seems to be an alternative: https://github.com/hyperspike/valkey-operator

(yet somewhat unmaintained recently 😐 )

**Reply by guimoreirar** [reply]
+1

**Reply by piterka** [reply]
+1 !
I've found also this: https://github.com/groundhog2k/helm-charts/tree/master/charts/valkey

**Reply by Mrmel94** [reply]
+1

**Reply by muffl0n** [reply]
Seems like this is the way to go now, isn't it? https://github.com/valkey-io/valkey-helm

**Reply by piterka** [reply]
Helm chart https://github.com/valkey-io/valkey-helm seems to be missing Sentinel support.

**Reply by mboncalo** [reply]
Hi Everyone,

Due to the same bitnami issue I created a Valkey helm chart for my company.
The helm chart uses valkey upstream images and valkey-sentinel
If anyone wants to test it, propose changes..I would  be happy to help. I have pushed it to my github:

https://github.com/mboncalo/valkey
